### ISSUE TITLE
PEP 585 error with TypedDict not recognized

### ISSUE THREAD

**Issue by cdce8p** [microsoft/pylance-release#917]
## Environment data

-   Language Server version: 2021.2.0
-   OS and version: darwin x64
-   Python version: **3.8**
-   python.analysis.indexing: true
-   python.analysis.typeCheckingMode: basic / strict

## Expected behaviour

Recognize `TypeError` for TypedDict equivalent syntax forms even with postponed evaluation enabled (Python **3.8**).
https://docs.python.org/3/library/typing.html#typing.TypedDict

## Actual behaviour

The code sample below causes two `TypeErrors` to be thrown even with `from __future__ import annotations`.
However, pylance only recognizes it if the future import is missing.

## Logs
--

## Code Snippet / Additional information

```python
from __future__ import annotations
from typing import TypedDict

CustomTypedDict = TypedDict("CustomTypedDict", my_var=list[int])  # TypeError
CustomTypedDict2 = TypedDict("CustomTypedDict2", {"my_var": list[int]})  # TypeError
```

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release.

**Comment by jakebailey** [comment]
This issue has been fixed in version 2021.2.1, which we've just released. You can find the changelog here: https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#202121-10-february-2021
